Description: Fixes for GTK3 compatibility
 wxWidgets 3.0 built with GTK3 doesn't have wxColour::GetPixel(), see:
 http://trac.wxwidgets.org/ticket/15141#comment:2
Author: Olly Betts <olly@survex.com>
Forwarded: no
Last-Update: 2018-04-01

--- a/XS/Colour.xs
+++ b/XS/Colour.xs
@@ -114,7 +114,7 @@
 
 #endif
 
-#if !defined(__WXMAC__)
+#if !defined(__WXMAC__) && !defined (__WXGTK3__)
 
 long
 wxColour::GetPixel()
--- a/ext/propgrid/XS/PGProperty.xsp
+++ b/ext/propgrid/XS/PGProperty.xsp
@@ -1148,7 +1148,7 @@
     bool SetPlValue( wxColour* col, int flags = wxPG_SETVAL_REFRESH_EDITOR )
       %code{%
             if( col->IsOk() ) {
-                wxVariant value = wxVariant(*col);
+                wxVariant value = wxVariant(wxAny(*col));
                 THIS->SetValue(value, NULL, flags );
                 RETVAL = true;
             } else {
